WeakAuras/Transmission.lua

```diff
@@ -730,454 +892,546 @@ local function scamCheck(codes, data)
+  -- through this table as well
+  init_started = true,
+  internalVersion = true,
+  controlledChildren = true,
+  parent = true,
+  expanded = true,
+}
+
+local deleted = {} -- magic value
+local function recurseDiff(ours, theirs)
+  local diff, seen, same = {}, {}, true
+  for key, ourVal in pairs(ours) do
+    if not ignoredForDiffChecking[key] then
+      seen[key] = true
+      local theirVal = theirs[key]
+      if type(ourVal) == "table" and type(theirVal) == "table" then
+        local diffVal = recurseDiff(ourVal, theirVal)
+        if diffVal then
+          diff[key] = diffVal
+          same = false
+        end
+      elseif ourVal ~= theirVal then
+        diff[key] = theirVal ~= nil and theirVal or deleted
+        same = false
+      end
     end
+  end
+  for key, theirVal in pairs(theirs) do
+    if not seen[key] and not ignoredForDiffChecking[key] then
+      diff[key] = theirVal
+      same = false
+    end
+  end
+  if not same then return diff end
+end
 
-    local regionType = data.regionType;
-    local regionData = regionOptions[regionType or ""]
-    local displayName = regionData and regionData.displayName or regionType or "";
+function WeakAuras.diff(ours, theirs)
+  -- generates a diff which WeakAuras.Update can use
+  if not ours or not theirs then return end
+  return recurseDiff(ours, theirs)
+end
 
-    local tooltip = {
-      {2, data.id, "          ", 0.5333, 0, 1},
-      {2, displayName, "          ", 1, 0.82, 0},
-      {1, " ", 1, 1, 1}
-    };
+function WeakAuras.MatchData(data, children)
+  -- setup
+  local info = {
+    unmodified = 0,
+    modified = 0,
+    added = 0,
+    deletedCount = 0,
+    deleted = {},
+    oldData = {},
+    diffs = {},
+    activeCategories = {},
+  }
 
-    local codes = {};
-    local highestVersion = data.internalVersion or 1;
-    scamCheck(codes, data);
-    if(children) then
-      for index, childData in pairs(children) do
-        tinsert(tooltip, {2, " ", childData.id, 1, 1, 1, 1, 1, 1});
-        scamCheck(codes, childData);
-        local childVersion = childData.internalVersion or 1;
-        highestVersion = max(highestVersion, childVersion);
-      end
-      if(#tooltip > 3) then
-        tooltip[4][2] = L["Children:"];
-      else
-        tinsert(tooltip, {1, L["No Children:"], 1, 1, 1});
-      end
-    elseif(data.controlledChildren) then
-      for index, childId in pairs(data.controlledChildren) do
-        tinsert(tooltip, {2, " ", childId, 1, 1, 1, 1, 1, 1});
-      end
-      if(#tooltip > 3) then
-        tooltip[4][2] = L["Children:"];
-      else
-        tinsert(tooltip, {1, L["No Children:"], 1, 1, 1});
+  -- match the parent/single aura (if no children)
+  local oldParent = WeakAuras.GetData(data.id)
+  if oldParent then
+    if data.uid and oldParent.uid and data.uid ~= oldParent.uid
+    or (children and (not oldParent.controlledChildren or oldParent.parent)) then
+      oldParent = nil
+    end
+  end
```

Comment:
This can match a standalone aura with a existing group if the id matches.

That is, create a group "New", import a aura named "New" and the offered dialog will offer to update the group with a standalone aura.

If the import has no children, then oldData should have no children either.